integration/library_test.go

```diff
@@ -174,96 +194,77 @@ func TestFanal_Library_TarMode(t *testing.T) {
 			ctx := context.Background()
 			d, _ := ioutil.TempDir("", "TestFanal_Library_")
 			defer os.RemoveAll(d)
-			c := cache.New(d)
+
+			c, err := cache.NewFSCache(d)
+			require.NoError(t, err)
+
+			applier := analyzer.NewApplier(c)
 
 			opt := types.DockerOption{
 				Timeout:  600 * time.Second,
 				SkipPing: true,
 			}
 
-			ext := docker.NewDockerExtractor(opt, c)
-			ac := analyzer.Config{Extractor: ext}
-			runChecks(t, "tar", ac, ctx, tc, d, nil, tc.imageFile)
+			ext, err := docker.NewDockerArchiveExtractor(ctx, tc.imageFile, opt)
+			require.NoError(t, err)
+			ac := analyzer.New(ext, c)
+			runChecks(t, ctx, ac, applier, tc)
 
 			// clear Cache
 			require.NoError(t, c.Clear(), tc.name)
 		})
 	}
 }
 
-func runChecks(t *testing.T, mode string, ac analyzer.Config, ctx context.Context, tc testCase, d string, c cache.Cache, testFile string) {
-	switch mode {
-	case "docker":
-		actualFiles, err := ac.Analyze(ctx, tc.imageFile)
-		require.NoError(t, err, tc.name)
-		commonChecks(t, actualFiles, tc)
-	case "tar":
-		actualFiles, err := ac.AnalyzeFile(ctx, testFile)
-		require.NoError(t, err, tc.name)
-		commonChecks(t, actualFiles, tc)
-	}
+func runChecks(t *testing.T, ctx context.Context, ac analyzer.Config, applier analyzer.Applier, tc testCase) {
+	imageInfo, err := ac.Analyze(ctx)
+	require.NoError(t, err, tc.name)
+	imageDetail, err := applier.ApplyLayers(imageInfo.LayerIDs)
+	require.NoError(t, err, tc.name)
+	commonChecks(t, imageDetail, tc)
 }
 
-func commonChecks(t *testing.T, actualFiles extractor.FileMap, tc testCase) {
-	checkFiles(t, actualFiles, tc)
-	osFound := checkOS(actualFiles, t, tc)
-	checkPackages(actualFiles, t, tc)
-	checkPackageFromCommands(t, actualFiles, osFound, tc)
-	checkLibraries(actualFiles, t, tc)
+func commonChecks(t *testing.T, detail types.ImageDetail, tc testCase) {
+	assert.Equal(t, tc.expectedOS, *detail.OS, tc.name)
+	checkPackages(detail, t, tc)
+	//checkPackageFromCommands(t, actualFiles, osFound, tc)
```

Comment:
We should bring this back but before we can, we need to figure out how we can get access to `actualFiles` again. `ImageDetail` no longer has this information.

Reply:
You're right. `PackageFromCommands` gets packages like the following.

```
RUN apk add sqlite-libs \
         && ./configure --with-sqlite \
         && make && make install \
         && apk del sqlite-libs
```

`sqlite-libs` is installed/removed in one layer, so we can't find it in a merged layer. But it is linked statically. Now, trivy detects vulnerabilities of `sqlite-libs` in this case.
This PR disables this feature. As you said, we have to figure out how we support this one. It makes this PR bigger. I'm supposed to implement it in a dedicated PR.